src/supremm/datasource/prometheus/prominterface.py

```diff
@@ -0,0 +1,476 @@
+import os
+import logging
+import urllib.parse as urlparse
+import datetime
+
+import numpy as np
+import requests
+
+from supremm.config import Config
+
+CHUNK_SIZE = 4 # HOURS
+
+
+class PromClient():
+    """ Client class to interface with Prometheus """
+
+    def __init__(self, resconf):
+        self._url = "http://{}".format(resconf['prom_host'])
+
+        self._client = requests.Session()
+        self._client.mount(self._url, self._client.get_adapter(self._url))
+        self._client.headers.update({'Content-Type': 'application/x-www-form-urlencoded',
+                                     'Accept': 'application/json'})
+
+        self._client.auth = (resconf['prom_user'], resconf['prom_password'])
+        self.connection = PromClient.build_info(self._client, self._url)
+
+    def __str__(self):
+        return self._url
+
+    @staticmethod
+    def build_info(client, test_url):
+        """ Query server build info. Test connection to server. """
+
+        endpoint = "/api/v1/status/buildinfo"
+        url = urlparse.urljoin(test_url, endpoint)
+
+        r = client.get(url)
+        if r.status_code != 200:
+            print(str(r.content))
+            return False
+
+        result = r.json()
+        if result["status"] == "success":
+            return True
+        else:
+            logging.warning("Unable to connect to Prometheus server at %s", url)
+            return False
+
+    def query(self, query, time):
+        """ Query an instantaneous value """
+
+        params = {
+            'query': query,
+            'time': time,
+        }
+
+        endpoint = "/api/v1/query"
+        url = urlparse.urljoin(self._url, endpoint)
+
+        r = self._client.get(url, params=params)
+        if r.status_code != 200:
+            print(str(r.content))
+            return None
+
+        return r.json()
+
+    def query_range(self, query, start, end):
+        """ Query a time range with a specified granularity """
+
+        params = {
+            'query': query,
+            'start': start,
+            'end': end,
+            'step': self._step
+        }
+
+        endpoint = "/api/v1/query_range"
+        url = urlparse.urljoin(self._url, endpoint)
+
+        r = self._client.get(url, params=params)
+        if r.status_code != 200:
+            print(r.content)
```

Comment:
Please use the logging class rather than `print`

Reply:
Resolved in https://github.com/ubccr/supremm/pull/259/commits/ba29736be72b3035bebd70ae7d0c636ac46ad87b